org.eclipse.jdt.ui.tests/ui/org/eclipse/jdt/ui/tests/quickfix/CleanUpTest1d5.java

```diff
@@ -4246,4 +4268,63 @@ public void testConvertStringBufferToStringBuilderAll() throws Exception {
 				new HashSet<>(Arrays.asList(MultiFixMessages.StringBufferToStringBuilderCleanUp_description)));
 	}
 
+	@Test
+	public void testOverrideMultiFix() throws Exception {
+		Hashtable<String, String> opts= JavaCore.getOptions();
+		opts.put(JavaCore.COMPILER_PB_MISSING_OVERRIDE_ANNOTATION, JavaCore.ERROR);
+		JavaCore.setOptions(opts);
+
+		IPackageFragment pack1= fSourceFolder.createPackageFragment("test1", false, null);
+		String sample= "" //
+				+ "package test1;\n" //
+				+ "public class E1 implements Comparable<String>, Runnable {\n" //
+				+ "    public void run() {};\n" //
+				+ "    public int compareTo(String o) { return -1; };\n" //
+				+ "}\n";
+		ICompilationUnit cu1= pack1.createCompilationUnit("E1.java", sample, false, null);
+
+		cu1.getJavaProject().getProject().build(IncrementalProjectBuilder.FULL_BUILD, null);
+
+		sample= "" //
+				+ "package test1;\n" //
+				+ "public class E1 implements Comparable<String>, Runnable {\n" //
+				+ "    @Override\n" //
+				+ "    public void run() {};\n" //
+				+ "    @Override\n" //
+				+ "    public int compareTo(String o) { return -1; };\n" //
+				+ "}\n";
+		String expected1= sample;
+
+		// Two error markers due to missing override annotations
+		IMarker[] markers= pack1.getResource().findMarkers(null, true, IResource.DEPTH_INFINITE);
+		assertEquals(2, markers.length);
+
+		IEditorInput input= EditorUtility.getEditorInput(cu1);
+		IProblemLocation location1= CorrectionMarkerResolutionGenerator.findProblemLocation(input, markers[0]);
+		IProblemLocation location2= CorrectionMarkerResolutionGenerator.findProblemLocation(input, markers[1]);
+
+		CleanUpOptions cleanUpOptions = new CleanUpOptions();
+		cleanUpOptions.setOption(CleanUpConstants.ADD_MISSING_ANNOTATIONS, CleanUpOptions.TRUE);
+		cleanUpOptions.setOption(CleanUpConstants.ADD_MISSING_ANNOTATIONS_OVERRIDE, CleanUpOptions.TRUE);
+		cleanUpOptions.setOption(CleanUpConstants.ADD_MISSING_ANNOTATIONS_OVERRIDE_FOR_INTERFACE_METHOD_IMPLEMENTATION, CleanUpOptions.TRUE);
+
+		Java50CleanUp cleanUp = new Java50CleanUp();
+		cleanUp.setOptions(cleanUpOptions);
+
+		ASTParser parser= ASTParser.newParser(IASTSharedValues.SHARED_AST_LEVEL);
+		parser.setResolveBindings(true);
+		parser.setProject(getProject());
+		final CompilationUnit[] roots= new CompilationUnit[1];
+		parser.createASTs(new ICompilationUnit[] { cu1 }, new String[0], new ASTRequestor() {
+			@Override
+			public void acceptAST(ICompilationUnit source, CompilationUnit ast) {
+				roots[0]= ast;
+			}
+		}, null);
```

Comment:
I think you can use `ASTNode createAST(IProgressMonitor monitor)` from `ASTParser`. Although it returns an `ASTNode`, you can check for yourself that it's always the `CompilationUnit` (ast root).

Reply:
Oh and you can use `parser.setSource(..)` to provide the `ICompilationUnit`

Reply:
Thanks!